Microsoft.Azure.Cosmos/src/Resource/Container/ContainerCore.Items.cs

```
@@ -1252,6 +1254,94 @@ private ChangeFeedProcessorBuilder GetChangeFeedProcessorBuilderPrivate(
+                            ranges: await parentFeedRangeInternal.GetEffectiveRangesAsync(
+                                routingMapProvider: routingMapProvider,
+                                containerRid: containerRId,
+                                partitionKeyDefinition: partitionKeyDefinition,
+                                trace: trace)),
+                        childRange: ContainerCore.GetRange(
+                            ranges: await childFeedRangeInternal.GetEffectiveRangesAsync(
+                                routingMapProvider: routingMapProvider,
+                                containerRid: containerRId,
+                                partitionKeyDefinition: partitionKeyDefinition,
+                                trace: trace)));
+                }
+                catch (DocumentClientException dce)
+                {
+                    throw CosmosExceptionFactory.Create(dce, trace);
+                }
+            }
+        }
+
+        private static Documents.Routing.Range<string> GetRange(
+            List<Documents.Routing.Range<string>> ranges)
+        {
+            if (ranges.Count == 1)
+            {
+                return ranges.First();
+            }
+
+            ranges.Sort(Documents.Routing.Range<string>.MinComparer.Instance);
+
+            Documents.Routing.Range<string> range = ranges.First();
+
+            return new Documents.Routing.Range<string>(
+                min: range.Min,
+                max: ranges.Last().Max,
+                isMinInclusive: range.IsMinInclusive,
+                isMaxInclusive: range.IsMaxInclusive);
+        }
+
+        private static bool IsSubset(
+            Documents.Routing.Range<string> parentRange,
+            Documents.Routing.Range<string> childRange)
+        {
+            return parentRange.Contains(childRange.Min)
+                && (parentRange.Max == childRange.Max || parentRange.Contains(childRange.Max));

```

Assuming maxEpk is same for parent and child for both then few combinations are still incorrectly returning subset right?

maxExclusive has four combinations 
True, True
True, False, 
False, False (Invalid but returns true)
False, True (Invalid but returns true)

---

For **False, False (Invalid but returns true)** and **False, True (Invalid but returns true)** we said a False isMinInclusive should throw the exception, but you say here (Invalid but returns true), but it can't throw the exception and return true, so what is the expected behavior here?

**True, False** is already tested in [GivenParentAndChildFeedRangesTrueIsMinInclusiveIsFeedRangePartOfAsyncTestAsync](https://github.com/Azure/azure-cosmos-dotnet-v3/blob/d5a999b2e85658717d26487dc5a9f96236b84fd0/Microsoft.Azure.Cosmos/tests/Microsoft.Azure.Cosmos.EmulatorTests/CosmosContainerTests.cs#L1906), and sure I can add **True, True** by adding that to the data row since this seems to be happy path.

So, it really goes back to **False, False (Invalid but returns true)** and **False, True (Invalid but returns true)**. Exception or return true?

---

Both the combinations are for maxExclusive only but for (parent, child)
The way to read it is (parentMaxInclusive, childMaxInclusive)

---

done.

---

> False, False (Invalid but returns true)
Actually this combination is Valid.

False, True is still invalid and should return false.

---

what is the logic behind this? or am I supposed to add the logic to look for that combination and fail it? By your own writing above, you say it should return true. Are we still not clear?

maxExclusive has four combinations
True, True
True, False,
False, False (Invalid but returns true)
**False, True (Invalid but returns true)**

---

Discussion requested @kirankumarkolli